pennylane/shadows/transforms.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classical shadow transforms"""
+
+from itertools import product
+from functools import reduce, wraps
+import warnings
+
+import pennylane as qml
+import pennylane.numpy as np
+
+
+@qml.batch_transform
+def _replace_obs(tape, obs, *args, **kwargs):
+    """
+    Tape transform to replace the measurement processes with the given one
+    """
+    # check if the measurement process of the tape is qml.classical_shadow
+    for o in tape.observables:
+        if o.return_type is not qml.measurements.Shadow:
+            raise ValueError(
+                f"Tape measurement must be {qml.measurements.Shadow!r}, got {o.return_type!r}"
```

Comment:
What does the `!r` do exactly?

Reply:
It calls `repr(qml.measurements.Shadow)` instead of `str(qml.measurements.Shadow)`, which is more descriptive.